digitiser-aggregator/src/main.rs

```diff
@@ -119,12 +127,8 @@ async fn on_message(
     producer: &FutureProducer,
     msg: &BorrowedMessage<'_>,
 ) {
-    if args.otel_endpoint.is_some() {
-        if let Some(headers) = msg.headers() {
-            debug!("Kafka Header Found");
-            OtelTracer::extract_context_from_kafka_to_span(headers, &tracing::Span::current());
-        }
-    }
+    msg.headers()
+        .conditional_extract_to_current_span(args.otel_endpoint.is_some());
```

Comment:
`_tracer.is_some()` is potentially a clearer test to use here (and in similar checks elsewhere).